app/models/ChargeState.js

```diff
@@ -0,0 +1,52 @@
+const logger = require('../utils/logger')(__filename)
+
+class ChargeState {
+  /**
+   * @param {Number} [createdAt]
+   * @param {Number} [accessedAt]
+   * @param {boolean} [isTerminal]
+   */
+  constructor (createdAt, accessedAt, isTerminal) {
+    this.createdAt = createdAt || epochSecondsNow()
+    this.accessedAt = accessedAt || epochSecondsNow()
+    this.isTerminal = isTerminal || false
+  }
+
+  updateAccessedAt () {
+    this.accessedAt = epochSecondsNow()
+  }
+
+  markTerminal () {
+    this.updateAccessedAt()
+    this.isTerminal = true
+  }
+
+  toString () {
+    return `${this.createdAt},${this.accessedAt},${this.isTerminal ? 'T' : 'F'}`
+  }
+}
+
+const chargeStateFromString = (data) => {
+  try {
+    const dataParts = data.split(',')
```

Comment:
What is this going to do if there’s no `data` (because it’s an old-format charge)?

Reply:
There’s a test that covers this scenario. The function returns a null value that is handled at the call site.

Reply:
Right you are.